udfs/tests/udf_test_utils.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+from os import path
+from os.path import dirname
+import re
+import argparse
+from google.cloud import bigquery
+
+from yaml import SafeLoader
+from yaml import load
+
+BIGQUERY_TEST_DATASET_MAPPINGS = {
+    'netezza': 'nz_test',
+    'oracle': 'or_test',
+    'redshift': 'rs_test',
+    'teradata': 'td_test',
+    'vertica': 've_test',
+    'community': 'fn_test',
+}
+
+UDF_PARENT_DIR = 'udfs/'
+
+
+def get_all_udf_paths():
+    return glob.glob(UDF_PARENT_DIR + '/**/*.sql', recursive=True)
+
+
+def load_test_cases(udf_path):
+    udf_dir = '/'.join(udf_path.split('/')[:-1])
+    yaml_test_data_path = f'{udf_dir}/test_cases.yaml'
+    if path.isfile(yaml_test_data_path):
+        with open(yaml_test_data_path, 'r') as yaml_file:
+            return load(yaml_file, Loader=SafeLoader)
+    else:
+        return None
+
+
+def extract_udf_name(udf_path):
```

Comment:
nit: test-ception
This logic in your testing framework utilities could use unittests.

Reply:
Addressed implicitly by adding a test to ensure that extract_udf_name is equivalent to the file name